api/pkg/webhooks/manager.go

```diff
@@ -0,0 +1,208 @@
+			tmpPayload = originalPayload
+		} else if tmpPayload, ok = responsePayloadLookup[client.GetUseDataFrom()]; !ok {
+			// This should only happen if a previous webhook had an error, but did not abort
+			// and the current client is trying to use the response from that client
+			return fmt.Errorf(
+				"webhook name %s not found, this could be because of an error in a previous webhook that this webhook depends on",
+				client.GetUseDataFrom(),
+			)
+		}
+		p, err := client.Invoke(ctx, tmpPayload)
+		if err == nil {
+			responsePayloadLookup[client.GetName()] = p
+			if client.IsFinalResponse() {
+				finalResponse = p
+			}
+			continue
+		}
+		// if err is not nil, check if client is set to abort
+		if client.AbortOnFail() {
+			return onError(err)
+		}
+
+	}
+	for _, client := range asyncClients {
+		// NOTE: Currently, this will never return err since InvokeAsync always returns nil
+		if err := client.InvokeAsync(ctx, originalPayload); err != nil {
+			return onError(err)
+		}
+	}
+	if err := onSuccess(finalResponse); err != nil {
+		// If the callback fails, return the error
+		return err
+	}
+
+	return nil
+}
+
+func parseAndValidateConfig(
+	eventList []EventType,
+	webhookConfigMap map[EventType][]WebhookConfig,
+) (WebhookManager, error) {
+	eventToWHMap := make(map[EventType]map[WebhookType][]WebhookClient)
+	for _, eventType := range eventList {
+		webhookConfigList, ok := webhookConfigMap[eventType]
+		if !ok {
+			continue
+		}
+		syncClients := make([]WebhookClient, 0)
+		asyncClients := make([]WebhookClient, 0)
+		for _, webhookConfig := range webhookConfigList {
+			if err := validateWebhookConfig(&webhookConfig); err != nil {
+				return nil, err
+			}
+			client := &simpleWebhookClient{
+				WebhookConfig: webhookConfig,
+			}
+			if !client.IsAsync() {
+				syncClients = append(syncClients, client)
+			} else {
+				asyncClients = append(asyncClients, client)
+			}
+		}
+		allClients := append(syncClients, asyncClients...)
+		if err := validateClients(allClients); err != nil {
+			return nil, err
+		}
+		tmpMap := make(map[WebhookType][]WebhookClient)
+		tmpMap[Async] = asyncClients
+		tmpMap[Sync] = syncClients
+		eventToWHMap[eventType] = tmpMap
+	}
+	return &SimpleWebhookManager{WebhookClients: eventToWHMap}, nil
+
+}
+
+// validateWebhookResponse ensures that the response from a webhook is either
+// a valid json object or empty str
+// This is only required for synchronous webhooks, where the webhook's response
+// may be used as input to the next webhook or in the callback function
+func validateWebhookResponse(content []byte) error {
+	if len(content) == 0 {
+		return nil
+	}
+	if json.Valid(content) {
+		return nil
+	}
+	return fmt.Errorf("webhook response is not a valid json object and not empty")
+}
+
+func validateClients(webhookClients []WebhookClient) error {
+	// ensure that only 1 sync client has finalResponse set to true
+	isFinalResponseSet := false
+	for _, client := range webhookClients {
```

Comment:
nit: do we need to loop through `webhookClients` thrice?

Reply:
Fixed